packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -1148,14 +1148,33 @@ export default class FulfillmentModuleService<
         shippingOption
       )
 
-      const existingRulesMap = new Map(
-        existingRules.map((rule) => [rule.id, rule])
-      )
+      const existingRulesMap: Map<
```

Comment:
Do you think it makes sense to use upsertWithReplace here?

Reply:
I would like to not spread that upsertWithReplace too much for now, because I believe we need to rework it when the DAL will be made properly replaceable and I believe for now this helper needs to be rethinked at a later point. since we are not in a place where there is many upsert with replace and it is pretty contained, I believe it is fine to keep it like it is. It was a bit different for the product where many of those actions was performed. is that ok for you? we can discuss more about it later if you want